# Motivation
Integration of heterogeneous and voluminous data from proteomics, transcriptomics, Immunological and clinical research constitutes not only a fundamental problem but a real hurdle in the extraction of valuable information to the surface from these omics data sets. The exponential increase of the novel omics technologies such as LC-MS/MS and the generation of high-resolution data from the large consortia projects generates heterogenous and big data sets. These big data promote the research and at the same time, new methodologies are lacking to analyze these complex data sets. The overall aim of my PhD research project is to carry out an integration of proteomic, transcriptomic, immunological and clinical datasets generated on individuals who have been exposed to Mycobacterium tuberculosis bacilli and who either have or have not become infected. More specifically, we have recruited a unique cohort of individuals who have either a latent tuberculosis infection, have had previous TB or who have had recurrent bouts of TB, as well as health care workers and household contacts of TB index cases who have remained long-term tuberculin skin test negative (i.e. presumed uninfected) and who may therefore have sterilizing immunity to TB. We have then made parallel proteomic, transcriptomic and immunology measurements on bronchoalveolar lavage (BAL) samples from the different groups of participants following directed infusion into the lungs of M. bovis BCG or purified protein derivative (of M. tuberculosis; ‘PPD’). This unique human lung challenge model of tuberculosis disease looks set to radically alter our understanding of immunological responses to exposure to M. tuberculosis, yet there are currently few bioinformatic tools available to enable data integration in this area.
